independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -103,49 +105,60 @@ static BeanInfo createProducerMethod(Set<Type> beanTypes, MethodInfo producerMet
                 // we needn't process it further, the annotation was a qualifier (or multiple repeating ones)
                 continue;
             }
-            if (DotNames.ALTERNATIVE.equals(annotation.name())) {
+            if (DotNames.ALTERNATIVE.equals(annotationName)) {
                 isAlternative = true;
                 continue;
             }
-            if (DotNames.ALTERNATIVE_PRIORITY.equals(annotation.name())) {
+            if (DotNames.ALTERNATIVE_PRIORITY.equals(annotationName)) {
                 isAlternative = true;
                 priority = annotation.value().asInt();
                 continue;
             }
             // This is not supported ATM but should work once we upgrade to Common Annotations 2.1
-            if ((!isAlternative || priority == null) && annotation.name().equals(DotNames.PRIORITY)) {
+            if ((!isAlternative || priority == null) && annotationName.equals(DotNames.PRIORITY)) {
                 priority = annotation.value().asInt();
                 continue;
             }
-            if (priority == null && DotNames.ARC_PRIORITY.equals(annotation.name())) {
+            if (priority == null && DotNames.ARC_PRIORITY.equals(annotationName)) {
                 priority = annotation.value().asInt();
                 continue;
             }
-            if (DotNames.DEFAULT_BEAN.equals(annotation.name())) {
+            if (DotNames.DEFAULT_BEAN.equals(annotationName)) {
                 isDefaultBean = true;
                 continue;
             }
-            ScopeInfo scopeAnnotation = beanDeployment.getScope(annotation.name());
+            ScopeInfo scopeAnnotation = beanDeployment.getScope(annotationName);
             if (scopeAnnotation != null) {
                 scopes.add(scopeAnnotation);
                 continue;
             }
-            StereotypeInfo stereotype = beanDeployment.getStereotype(annotation.name());
+            StereotypeInfo stereotype = beanDeployment.getStereotype(annotationName);
             if (stereotype != null) {
                 stereotypes.add(stereotype);
                 continue;
             }
+            BeanDefiningAnnotation bda = beanDeployment.getBeanDefiningAnnotation(annotationName);
+            if (bda != null && bda.getDefaultScope() != null) {
+                beanDefiningAnnotationScopes.add(beanDeployment.getScope(bda.getDefaultScope()));
+            }
```

Comment:
This code still won't execute if the BDA is also a qualifier, no? Hence the code below to find the scope of the bean won't see the default scope of the BDA, I think?

Reply:
Yes, you're right. My test is wrong!